### PR TITLE
Game is draw

### PR DESCRIPTION
**Pull request by goformarty**
- isBoardFull() method added to board component
- new constructor function creating gird array added to board component
- board component generally refactored and extra empty lines added for clarity
- game component takes board component as a provider
- game component constructor updated to NOT start with a new instance of board component
- isDraw() method added to game component

### CODE REVIEW

File: src/app/board/board.component.spec.ts [hunk 1]
```diff
@@ -24,22 +24,50 @@ describe('BoardComponent', () => {
...
+    it('an empty board that is not full', () => {
+      const board = new BoardComponent();
+
+      expect(board.isBoardFull()).toEqual(false);
+    });
+
+    it('creates a full board', () => {
+      const board = SpecHelper.setupBoard('X', 'X', 'X', 'O', 'X', 'X', 'X', 'X', 'X');
+
+      expect(board.isBoardEmpty()).toEqual(false);
```
Comment (cnak) [inline src/app/board/board.component.spec.ts]: Perhaps this was a typo? I think you wanted to test whether isBoardFull returns true.
In addition you've already tested the isboardempty method

File: src/app/board/board.component.ts [hunk 2]
```diff
@@ -8,14 +8,29 @@ import { SpecHelper } from '../spec-helper/spec.helper';
...
+  isBoardFull(): boolean {
     return (this.remainingMoves() === 0) ? true : false;
   }
 
+
+  isValid(position: number): boolean {
+    return (this.isValidRange(position) && this.isPositionEmpty(position));
+  }
+
+  printBoard() {
```
Comment (cnak) [inline src/app/board/board.component.ts]: Please remove this was part of a quick spike

File: src/app/board/board.component.ts [hunk 3]
```diff
@@ -26,16 +41,21 @@ export class BoardComponent {
...
+  isBigEnough(element, index, array) {
+    console.log(element);
+   return (element !== undefined );
   }
 
-  private isValid(position: number): boolean {
-    return (this.isValidRange(position) && this.isPositionEmpty(position));
+  // remainingMoves(): number {
+  //   return this.grid.filter(value => value === undefined).length;
+  // }
```
Comment (cnak) [inline src/app/board/board.component.ts]: Don't commit commented out code

File: src/app/board/board.component.ts [hunk 4]
```diff
@@ -26,16 +41,21 @@ export class BoardComponent {
...
   }
 
-  private remainingMoves() {
-    const remaining = this.grid.filter(
-      function (value) {
-        return value === undefined;
-      });
-    return remaining.length;
+  isBigEnough(element, index, array) {
+    console.log(element);
```
Comment (cnak) [inline src/app/board/board.component.ts]: Remove this unused method

**Review by cnak — CHANGES_REQUESTED** [review]
Please review

**Comment by goformarty** [comment]
- print board methods created for spike: removed
- unnecessary commented out code: removed
- test for full board expecting true corrected

**Pull request merged**